core/src/main/java/org/elasticsearch/index/query/GeoBoundingBoxQueryBuilder.java

```diff
@@ -19,78 +19,140 @@
+import org.elasticsearch.index.search.geo.IndexedGeoBoundingBoxQuery;
 
 import java.io.IOException;
+import java.util.Objects;
 
+/**
+ * Creates a Lucene query that will filter for all documents that lie within the specified
+ * bounding box.
+ *
+ * This query can only operate on fields of type geo_point that have latitude and longitude
+ * enabled.
+ * */
 public class GeoBoundingBoxQueryBuilder extends AbstractQueryBuilder<GeoBoundingBoxQueryBuilder> {
-
+    /** Name of the query. */
     public static final String NAME = "geo_bbox";
+    /** Default for geo point coerce (as of this writing false). */
+    public static final boolean DEFAULT_COERCE = false;
+    /** Default for skipping geo point validation (as of this writing false). */
+    public static final boolean DEFAULT_IGNORE_MALFORMED = false;
+    /** Default type for executing this query (memory as of this writing). */
+    public static final Type DEFAULT_TYPE = Type.MEMORY;
+    /** Needed for serialization. */
+    static final GeoBoundingBoxQueryBuilder PROTOTYPE = new GeoBoundingBoxQueryBuilder();
+
+    /** Name of field holding geo coordinates to compute the bounding box on.*/
+    private final String fieldName;
+    /** Top left corner coordinates of bounding box. */
+    private GeoPoint topLeft = new GeoPoint(Double.NaN, Double.NaN);
+    /** Bottom right corner coordinates of bounding box.*/
+    private GeoPoint bottomRight = new GeoPoint(Double.NaN, Double.NaN);
+    /** Whether or not to infer correct coordinates for wrapping bounding boxes.*/
+    private boolean coerce = DEFAULT_COERCE;
+    /** Whether or not to skip geo point validation. */
+    private boolean ignoreMalformed = DEFAULT_IGNORE_MALFORMED;
+    /** How the query should be run. */
+    private Type type = DEFAULT_TYPE;
 
-    public static final String TOP_LEFT = GeoBoundingBoxQueryParser.TOP_LEFT;
-    public static final String BOTTOM_RIGHT = GeoBoundingBoxQueryParser.BOTTOM_RIGHT;
-
-    private static final int TOP = 0;
-    private static final int LEFT = 1;
-    private static final int BOTTOM = 2;
-    private static final int RIGHT = 3;
-
-    private final String name;
-
-    private double[] box = {Double.NaN, Double.NaN, Double.NaN, Double.NaN};
-
-    private String type;
-    private Boolean coerce;
-    private Boolean ignoreMalformed;
-
-    static final GeoBoundingBoxQueryBuilder PROTOTYPE = new GeoBoundingBoxQueryBuilder(null);
+    /**
+     * For serialization only.
+     * */
+    private GeoBoundingBoxQueryBuilder() {
+        this.fieldName = null;
+    }
 
-    public GeoBoundingBoxQueryBuilder(String name) {
-        this.name = name;
+    /**
+     * Create new bounding box query.
+     * @param fieldName name of index field containing geo coordinates to operate on.
+     * */
+    public GeoBoundingBoxQueryBuilder(String fieldName) {
+        if (fieldName == null || ("".equals(fieldName) != false)) {
```

Comment:
Oh - I missed that this class exists in our code base instead of only in Google Guava. (Stayed away from it due to https://github.com/elastic/elasticsearch/issues/13224 )